src/iotjs.c

```diff
@@ -216,15 +219,37 @@ int iotjs_entry(int argc, char** argv) {
   int res = uv_loop_close(iotjs_environment_loop(env));
   IOTJS_ASSERT(res == 0);
 
+  // Check whether context reset was sent or not.
+  if (iotjs_environment_config(env)->debugger != NULL) {
+    jerry_value_t res;
+    jerry_debugger_wait_for_source_status_t receive_status;
+    receive_status =
+        jerry_debugger_wait_for_client_source(dummy_wait_for_client_source_cb,
+                                              NULL, &res);
+
+    if (receive_status == JERRY_DEBUGGER_CONTEXT_RESET_RECEIVED) {
+      iotjs_environment_config(env)->debugger->context_reset = true;
+    }
+    jerry_release_value(res);
+  }
+
   // Release JerryScript engine.
   iotjs_jerry_release(env);
 
-terminate:
+terminate:;
```

Comment:
Leaving `;` here looks strange. I also failed to notice it at my first glance. Do we surely need this?

Reply:
You must have a statement after a goto label, and variable declaration isn't one, so ; counts as an empty statement.